sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/DefaultNetworkSpec.java

```diff
@@ -2,29 +2,81 @@
 
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.mesosphere.sdk.specification.validation.ValidationUtils;
 import org.apache.commons.lang3.builder.EqualsBuilder;
 import org.apache.commons.lang3.builder.HashCodeBuilder;
 
-import javax.validation.constraints.NotNull;
-import javax.validation.constraints.Size;
+import javax.validation.Valid;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
 
 /**
- * Default implementation of {@link NetworkSpec}.
+ * Default implementation of {@link NetworkSpec}. This class encapsulates the Container Network Interface
+ * (CNI) implementation within the SDK.
  */
 public class DefaultNetworkSpec implements NetworkSpec {
-    @NotNull
-    @Size(min = 1)
-    private String name;
+    @Valid
+    private String networkName;  // name of the overlay network to join, usually 'dcos'
```

Comment:
This documentation is useful, but maybe instead of on the private members, have it be part of a javadoc for the constructor? This is useful both because it'll end up in our class documentation tree and because IMO it subtly encourages consumers of this class to think in terms of interface rather than implementation in terms of how they consume it (e.g. for extension, or what have you).